flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -172,13 +173,20 @@
 
 	/** The library cache, from which the task can request its required JAR files */
 	private final LibraryCacheManager libraryCache;
-	
+
 	/** The cache for user-defined files that the invokable requires */
 	private final FileCache fileCache;
-	
+
 	/** The gateway to the network stack, which handles inputs and produced results */
 	private final NetworkEnvironment network;
 
+	/** The registry of this task which enables live reporting of accumulators */
+	private final AccumulatorRegistry accumulatorRegistry;
```

Comment:
Since the AccumulatorRegistry is only used task-internally, and always retrieved form there, it should be initialized internally. Saves one more constructor parameter and helps with separation of concerns.